lib/marc_cleanup/record_level.rb

```diff
@@ -13,16 +15,55 @@ def bad_utf8?(record)
     record.to_s.scrub != record.to_s
   end
 
+  def bad_utf8_scrub_datafield(record:, field:, field_index:)
+    field.subfields.each_with_index do |subfield, subfield_index|
+      new_value = bad_utf8_scrub_value(subfield.value)
+      record.fields[field_index].subfields[subfield_index].value = new_value
+    end
+    record
+  end
+
+  def bad_utf8_scrub_value(string)
+    string.scrub('').force_encoding('UTF-8')
+  end
+
+  ### Scrub invalid UTF-8 byte sequences within field values,
+  #     replacing with nothing; indicators, subfield codes, and tags must be
+  #     handled separately
+  def bad_utf8_scrub(record)
+    record.fields.each_with_index do |field, field_index|
+      if field.instance_of?(MARC::DataField)
+        record = bad_utf8_scrub_datafield(record: record,
+                                          field: field,
+                                          field_index: field_index)
```

Comment:
To me, it would be clearer if `bad_utf8_scrub_datafield` returned a new `MARC::DataField`, rather than a `MARC::Record`.  While I think it's fine in this case, it can lead to confusing bugs when you are mutating the same object that you are iterating through (in this case `record`).